classes/Customer.php

```diff
@@ -197,7 +197,11 @@ class CustomerCore extends ObjectModel
 
     public function __construct($id = null)
     {
-        $this->id_default_group = (int)Configuration::get('PS_CUSTOMER_GROUP');
```

Comment:
I think you should go back to the root of the problem : why is tax display looking at CustomerCore::id_default_group ? This property is used as a fallback when creating a customer group. This is not customer data. It tells nothing about currently instantiated Customer object.
You should investigate the code responsible for tax display, and find another way to determine if we need to display taxes or not.